.github/workflows/python-app.yml

```diff
@@ -7,59 +7,63 @@ on:
     branches: [ master ]
 
 jobs:
-  build:
-
+  Validate:
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v2
+
+    - name: Checkout
+      uses: actions/checkout@v4
       with:
-        python-version: 3.7
+        # download from the PR's repo (which may be a fork), because we also
+        # need the git-annex branch from that repo
+        repository: ${{github.event.pull_request.head.repo.full_name}}
+        ref: ${{github.event.pull_request.head.ref}}
```

Comment:
This syntax [looks good to me](https://github.com/actions/checkout/issues/551#issuecomment-1113089390)?

Reply:
Thanks for the reference. I got it from here myself, but didn't check its lineage:
https://github.com/spine-generic/data-multi-subject/blob/2a9ac1c352d8a0c40bb2ee6b025aaa72eadc73b2/.github/workflows/validator.yml#L25-L35